### PR TITLE
fix: GridForm add validation type to GridFormHiddenField to fix failing compile

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

image

[fixed w/ alpha build
](https://github.com/codecademy-engineering/Codecademy/pull/24135/checks?check_run_id=2511725496)

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/GridForm/types.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ type HiddenField = Omit<BaseFormField<any>, 'size'>;
 
 export type GridFormHiddenField = HiddenField & {
   type: 'hidden';
+  validation?: ValidationRules;
```
Comment (cstatro) [inline packages/gamut/src/GridForm/types.ts:108]: do we want this be `never` instead ?

**Review by cstatro — APPROVED** [review]

File: packages/gamut/src/GridForm/types.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/GridForm/types.ts:108]: Oh yeah @dreamwasp I think the issue is that the test in the monolith shouldn't assume this exists. I like it most as-is :)

**Review by codecaaron — APPROVED** [review]

File: packages/gamut/src/GridForm/types.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/GridForm/types.ts:108]: Hmm yeah I like it more the way it is now. We would never want to add a validation to the hidden fields. IMO the real issue is that the monolith test assumes this exists, and should probably use a type assertion instead.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Sorry to be a contrarian 😞

File: packages/gamut/src/GridForm/types.ts (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/GridForm/types.ts:108]: all good - just to be clear, what you want me to do is fix the monolith test that is causing the compiler error?

**Pull request closed without merging**